lib/datadog/tracing/contrib/redis/instrumentation.rb

```diff
@@ -19,51 +19,58 @@ def self.included(base)
           # InstanceMethods - implementing instrumentation
           module InstanceMethods
             def call(*args, &block)
-              response = nil
+              show_command_args = command_args?
+
               Tracing.trace(Contrib::Redis::Ext::SPAN_COMMAND) do |span|
-                span.service = Datadog.configuration_for(redis_instance, :service_name) ||
-                  datadog_configuration[:service_name]
+                span.service = service_name
                 span.span_type = Contrib::Redis::Ext::TYPE
-                span.resource = get_command(args)
-                Contrib::Redis::Tags.set_common_tags(self, span)
-                span.set_tag(Tracing::Metadata::Ext::TAG_KIND, Tracing::Metadata::Ext::SpanKind::TAG_CLIENT)
+                span.resource = get_command(args, show_command_args)
+                Contrib::Redis::Tags.set_common_tags(self, span, show_command_args)
 
-                response = super
+                super
               end
-
-              response
             end
 
             def call_pipeline(*args, &block)
-              response = nil
+              show_command_args = command_args?
+
               Tracing.trace(Contrib::Redis::Ext::SPAN_COMMAND) do |span|
-                span.service = Datadog.configuration_for(redis_instance, :service_name) ||
-                  datadog_configuration[:service_name]
+                span.service = service_name
                 span.span_type = Contrib::Redis::Ext::TYPE
-                commands = get_pipeline_commands(args)
+                commands = get_pipeline_commands(args, show_command_args)
                 span.resource = commands.any? ? commands.join("\n") : '(none)'
                 span.set_metric Contrib::Redis::Ext::METRIC_PIPELINE_LEN, commands.length
-                Contrib::Redis::Tags.set_common_tags(self, span)
-                span.set_tag(Tracing::Metadata::Ext::TAG_KIND, Tracing::Metadata::Ext::SpanKind::TAG_CLIENT)
+                Contrib::Redis::Tags.set_common_tags(self, span, show_command_args)
 
-                response = super
+                super
               end
-
-              response
             end
 
             private
 
-            def get_command(args)
-              if datadog_configuration[:command_args]
+            def command_args?
+              pinned = Datadog.configuration_for(redis_instance, :command_args)
```

Comment:
`Datadog.configuration_for(redis_instance, :command_args)` is not a trivial operation: I recommend performing it only one per `Redis#call` or `Redis#call_pipeline` by calling it outside of `command_args?`. I recommend storing the result of `config = Datadog.configuration_for(redis_instance, :command_args)` and passing it to both `command_args?` and `service_name`.